docs/blog/_posts/2020-03-18-23rd-dotty-milestone-release.md

```diff
@@ -0,0 +1,289 @@
+```
+
+We hope that this change will improve the readability of context functions for a person who already knows the syntax for context parameters of ordinary methods.
+
+## Drop `given` parameter syntax
+As part of our experimentation with the syntax of the language, we are now dropping the old syntax for context parameters.
+
+The old syntax for context parameters was as follows:
+
+```scala
+// OLD SYNTAX, NO LONGER SUPPORTED
+def f(given x: Int) = x * x
+```
+
+In the previous release, it was [replaced](https://dotty.epfl.ch/blog/2020/02/05/22nd-dotty-milestone-release.html#further-improvements-to-the-context-parameters-syntax) by the new `using` syntax:
+
+```scala
+def f(using x: Int) = x * x
+```
+
+However, both syntaxes were supported for that release for experimental purposes. Now, we are dropping the support of the old syntax in favor of the new one as we see it as a clear win over the old one.
+
+# Metaprogramming
+## Inline version of `summon`
+When working from within an inline method, it is often the case that we want to summon a value without declaring it as a context parameter of the method:
+
+```scala
+inline def lookup[X] =
+  val x = summon[X]  // error
+  // -- Error: /Users/kmetiuk/Projects/scala3/pg/release/snip_5.scala:6:19 ----------
+  // 6 |  val x = summon[X]
+  //   |                   ^
+  //   |no implicit argument of type X was found for parameter x of method summon in object DottyPredef
+  // 1 error found
+  println(s"x = $x")
+```
+
+The above program will give us a compile time error because it cannot find a context parameter of type `X`. This is because the `summon` function is not inline and hence the compiler needs to know that a context parameter of type `X` exists on call site of `summon` which happens to be in the body of `lookup`. Since `X` is unknown in that body, the compiler can't find the context parameter and shows an error.
+
+We have now added an inline version of `summon`:
+
+```scala
+import scala.compiletime.summonInline
+
+inline def lookup[X] =
+  val x = summonInline[X]
+  println(s"x = $x")
+
+@main def Test =
+  given Int = 10
+  lookup[Int]
+```
+
+`summonInline` is an inline version of `summon`. It is defined as follows:
+
+```scala
+inline def summonInline[T] <: T = summonFrom {
+  case t: T => t
+}
+```
+
+Since it is inline, the context parameter is resolved on expansion site, not the call site. The expansion site happens to be wherever `lookup` function is expanded, and there the type `X` is bound to a concrete type.
+
+## `ValueOfExpr` renamed to `Unlifted`
+This feature allows you to obtain the value captured in an expression using pattern matching:
+
+Macro.scala
+
+```scala
+import scala.quoted._
+
+inline def square(x: Int): Int = ${ squareImpl('x) }
+def squareImpl(x: Expr[Int])(using QuoteContext): Expr[Int] =
+  x match
+    case Unlifted(value: Int) => Expr(value * value)
+```
+
+Test.scala
+
+```scala
+@main def Test =
+  println(square(10))  // 100
```

Comment:
```suggestion
  println(square(10))  // println(100)
```

Reply:
In the comment, there's the runtime output that should be observed by the user. Not the result of the inlining which the user cannot observe.

Reply:
But here the output is the code which is the part that is interesting to see